src/main/scala/edu/berkeley/ce/rockslicing/RockSlicer.scala

```diff
@@ -10,24 +10,36 @@ object RockSlicer {
   def main(args: Array[String]) {
     val conf = new SparkConf().setAppName("SparkRocks")
     val sc = new SparkContext(conf)
-    val arguments = CommandReader.parseArguments(args)
+    val parsedArgs = CommandReader.parseArguments(args)
+    if (parsedArgs.isEmpty) {
+      // Error message will already be printed out by CommandReader
+      System.exit(-1)
+    }
+    val arguments = CommandReader.parseArguments(args).get
 
     // Open and read input file specifying rock volume and joints
-    val inputSource = Source.fromFile(arguments('inputFile).toString)
-    val (rockVolume, joints) = InputProcessor.readInput(inputSource)
+    val inputSource = Source.fromFile(arguments.inputFile)
+    val inputOpt = InputProcessor.readInput(inputSource)
     inputSource.close()
-    var blocks = Vector(Block((0.0, 0.0, 0.0), rockVolume))
+    if (inputOpt.isEmpty) {
+      // Error message will already be printed out by InputProcessor
+      System.exit(-1)
+    }
+    val (rockVolume, joints) = inputOpt.get
+    var blocks = Seq(Block((0.0, 0.0, 0.0), rockVolume))
```

Comment:
Yes, I think so.

So this won't really be a huge change then, right? Just reading in some more information from the input file and doing the relevant error checking, and using this information to initialize our rock mass. I'll go ahead and implement this.